src/Microsoft.ComponentDetection/Program.cs

```diff
@@ -2,52 +2,70 @@
+using Serilog.Events;
+using Spectre.Console.Cli;
 
-try
+if (args.Contains("--debug", StringComparer.OrdinalIgnoreCase))
 {
-    if (args.Contains("--debug", StringComparer.OrdinalIgnoreCase))
+    Console.WriteLine($"Waiting for debugger attach. PID: {Process.GetCurrentProcess().Id}");
+    while (!Debugger.IsAttached)
     {
-        Console.WriteLine($"Waiting for debugger attach. PID: {Process.GetCurrentProcess().Id}");
-        while (!Debugger.IsAttached)
-        {
-            await Task.Delay(1000);
-        }
+        await Task.Delay(1000);
     }
+}
 
-    Log.Logger = new LoggerConfiguration()
-        .WriteTo.Console(outputTemplate: "[BOOTSTRAP] [{Timestamp:HH:mm:ss} {Level:u3}] {Message:lj}{NewLine}{Exception}")
-        .CreateBootstrapLogger();
-
-    var serviceProvider = new ServiceCollection()
-        .AddComponentDetection()
-        .ConfigureLoggingProviders()
-        .BuildServiceProvider();
-    var orchestrator = serviceProvider.GetRequiredService<Orchestrator>();
-    var result = await orchestrator.LoadAsync(args);
+var serviceCollection = new ServiceCollection()
+    .AddComponentDetection()
+    .AddLogging(l => l.AddSerilog(new LoggerConfiguration()
+        .MinimumLevel.ControlledBy(LogLevelInterceptor.LogLevel)
+        .Enrich.With<LoggingEnricher>()
+        .WriteTo.Map(
+            LoggingEnricher.LogFilePathPropertyName,
+            (logFilePath, wt) => wt.Async(x => x.File($"{logFilePath}")),
+            1) // sinkMapCountLimit
+        .WriteTo.Map<bool>(
+            LoggingEnricher.PrintStderrPropertyName,
+            (printLogsToStderr, wt) => wt.Console(standardErrorFromLevel: printLogsToStderr ? LogEventLevel.Debug : null),
+            1) // sinkMapCountLimit
+        .CreateLogger()));
 
-    var exitCode = (int)result.ResultCode;
-    if (result.ResultCode is ProcessingResultCode.Error or ProcessingResultCode.InputError)
+using var registrar = new TypeRegistrar(serviceCollection);
+var app = new CommandApp<ListDetectorsCommand>(registrar);
+app.Configure(
+    config =>
     {
-        exitCode = -1;
-    }
+        var resolver = registrar.Build();
 
-    Console.WriteLine($"Execution finished, status: {exitCode}.");
+        // Create the logger here as the serviceCollection will be disposed by the time we need to use the exception handler.
+        var logger = resolver.Resolve(typeof(ILogger<Program>)) as ILogger<Program>;
 
-    // Manually dispose to flush logs as we force exit
-    await serviceProvider.DisposeAsync();
+        config.SetInterceptor(new Interceptor(resolver, resolver.Resolve(typeof(ILogger<Interceptor>)) as ILogger<Interceptor>));
+        config.SetInterceptor(new LogLevelInterceptor());
+        config.SetInterceptor(new PrintManifestInterceptor());
 
-    await Log.CloseAndFlushAsync();
+        config.Settings.ApplicationName = "component-detection";
 
-    // force an exit, not letting any lingering threads not responding.
-    Environment.Exit(exitCode);
```

Comment:
Do we need to handle threads properly with the new approach? Just double checking we are freeing all resources before we terminate the tool's execution.

Reply:
The `ServiceCollection` is disposed inside of the `TypeRegistrar`.

Additionally, each `Command` returns an exit code, which is propagated up by `Spectre.Console.Cli` to the `RunAsync`, which is returned.

Reply:
Also verify the exit code remains the same based on the internal values we are giving it from the orchestrator execution.

Reply:
> Also verify the exit code remains the same based on the internal values we are giving it from the orchestrator execution.

This will have to be done in internal, as it has it's own separate orchestrator that will be migrated seperately.